### PR TITLE
Adding -Extension and -LeafBase switches to Split-Path

### LINKED ISSUE #2700: Split-Path should support -Extension and -LeafBase
$f = "c:\temp\foo.txt"

Split-Path $f -Leaf -> "foo.txt"
Split-Path $f -Extension -> ".txt"  (with the "." to be compatible with dir)
Split-Path $f -LeafBase -> "foo"

### PR DESCRIPTION
**Pull request by powercode**
Fixes #2700 

Extension and LeafBase are specializations of Leaf, and uses System.IO.Path.GetExtension
and System.IO.Path.GetFilenameWithoutExtension to extract parts from the Leaf

Tests added in test\powershell\Modules\Microsoft.PowerShell.Management\FileSystem.Tests.ps1

### CODE REVIEW

File: test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1 [hunk 1]
```diff
@@ -577,6 +578,16 @@ Describe "Extended FileSystem Path/Location Cmdlet Provider Tests" -Tags "Featur
...
+        
+        It 'Validate LeafBase' {
+            $result = Split-Path -Path "$level2_1Full$fileExt" -LeafBase
+            $result | Should Be $level2_1
+        }
+
+        It 'Validate LeafBase' {
+            $result = Split-Path -Path "$level2_1Full$fileExt" -Extension
+            $result | Should Be $fileExt
+        }
```
Comment (JamesWTruher) [inline test/powershell/Modules/Microsoft.PowerShell.Management/FileSystem.Tests.ps1]: should there be explicit tests for things like:
"$level2_1Full$fileExt$fileExt"
to ensure that the trimming is not over-zealous?

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ParsePathCommand.cs [hunk 2]
```diff
@@ -37,6 +39,17 @@ public class SplitPathCommand : CoreCommandWithCredentialsBase
         private const string leafSet = "LeafSet";
 
         /// <summary>
+        /// The parameter set name to get the leaf name
+        /// </summary>
+        private const string leafBaseSet = "LeafBaseSet";
```
Comment (charub) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ParsePathCommand.cs]: What's the difference between LeafSet and LeafBaseSet?

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ParsePathCommand.cs [hunk 3]
```diff
@@ -37,6 +39,17 @@ public class SplitPathCommand : CoreCommandWithCredentialsBase
...
+
+        /// <summary>
+        /// The parameter set name to get the leaf name
```
Comment (charub) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ParsePathCommand.cs]: The comment seems to be duplicated.

File: src/Microsoft.PowerShell.Commands.Management/commands/management/ParsePathCommand.cs [hunk 4]
```diff
@@ -154,18 +135,7 @@ public SwitchParameter NoQualifier
         /// </value>
         ///
         [Parameter(ParameterSetName = parentSet, Mandatory = false, ValueFromPipelineByPropertyName = true)]
-        public SwitchParameter Parent
```
Comment (charub) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ParsePathCommand.cs]: The default value of _parent used to be true. The default for a switch is False. Please revisit.

Comment (powercode) [inline src/Microsoft.PowerShell.Commands.Management/commands/management/ParsePathCommand.cs]: Good catch. Fixing and rebasing.

**Comment by daxian-dbw** [comment]
@charub your comments have been addressed, can you please take another look? If the change looks good, please approve the PR.

**Review by daxian-dbw — APPROVED** [review]

**Pull request merged**